File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l=0,m=0,h=nums.size()-1;
        while(m<=h)
        {
            if(nums[m]==0)
            {
                swap(nums[m],nums[l]);
                m++,l++;
            }
            else if(nums[m]==2)
            {
                swap(nums[m],nums[h]);
                h--;
            }
            else if(nums[m]==1)
            {
                m++;
            }
        }
        
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    protected void swap(int []nums,int a,int b)
    {           nums[a]=nums[a]^nums[b];
                nums[b]=nums[a]^nums[b];
                nums[a]=nums[a]^nums[b]; 
    }
    public void sortColors(int[] nums) {
        int left =0,high =nums.length-1,mid=0;
         while(mid<=high)
         {
            if(nums[mid]==0 && mid!=left){
                swap(nums,left,mid);
                left++;
            }
            else if(nums[mid]==2 && mid!=high){
                swap(nums,high,mid);
                high--;
            }
            else mid++;
         }
        
    }
}